docs/guides/timelock-proposal.md

```diff
@@ -0,0 +1,333 @@
+    }
+
+    // Validates the post-execution state.
+    function _validate(Addresses addresses, address) internal override {
+        address timelock = addresses.getAddress("PROTOCOL_TIMELOCK");
+        Vault timelockVault = Vault(addresses.getAddress("VAULT"));
+        MockToken token = MockToken(addresses.getAddress("TOKEN_1"));
+
+        assertEq(timelockVault.owner(), timelock);
+        assertTrue(timelockVault.tokenWhitelist(address(token)));
+        assertFalse(timelockVault.paused());
+
+        assertEq(token.owner(), timelock);
+        assertEq(token.balanceOf(timelock), token.totalSupply());
+    }
+}
+```
+
+Let's go through each of the functions we are overriding here.
+
+-   `name()`: Define the name of your proposal.
+-   `description()`: Provide a detailed description of your proposal.
+-   `_deploy()`: Deploy any necessary contracts. This example demonstrates the deployment of Vault and an ERC20 token.
+-   `_build(`): Set the necessary actions for your proposal. In this example, ERC20 token is whitelisted on the Vault contract
+-   `_run()`: Execute the proposal actions outlined in the `_build()` step. This
+    function performs a call to `_simulateActions` from the inherited
+    `TimelockProposal` contract. Internally, `_simulateActions()` simulates a call to Timelock [excheduleBatch](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/TimelockController.sol#L291) and [executeBatch](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/TimelockController.sol#L385) with the calldata generated from the actions set up in the build step.
+-   `_validate(`): This final step is crucial for validating the post-execution state. It ensures that the timelock is the new owner of Vault and token, the tokens were transferred to timelock and the token was whitelisted on the Vault contract
+
+Now that your first proposal contract is ready, it's time to take action. You have two options to execute the contract. The first option is to use `foundry test`. You can learn how to do that on [integration-tests.md](../testing/integration-tests.md "mention") section. The second option is to use `foundry script`, which is the method we will use here. But first, we need to set up [Addresses](../overview/architecture/addresses.md) contract. Let's create a `Addresses.json` file:
+
+```json
+[
+    {
+        "addr": "0x1a9C8182C09F50C8318d769245beA52c32BE35BC",
+        "name": "PROTOCOL_TIMELOCK",
+        "chainId": 31337
+    },
+    {
+        "addr": "0x10A19e7eE7d7F8a52822f6817de8ea18204F2e4f",
+        "name": "TIMELOCK_PROPOSER",
+        "chainId": 31337
+    },
+    {
+        "addr": "0x10A19e7eE7d7F8a52822f6817de8ea18204F2e4f",
+        "name": "TIMELOCK_EXECUTOR",
+        "chainId": 31337
+    }
+]
+```
+
+Now that we have the JSON file to be used on `Addresses.sol`, let's create a script that inherits `ScriptSuite`.
+
+```solidity
```

Comment:
What do we think about moving these code examples into the `examples/` directory?